noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -501,7 +500,9 @@ impl PrivateContext {
                 historical_header: Header::empty(),
                 prover_address: AztecAddress::zero(),
                 revert_code: 0,
-                gas_left: Gas::empty()
+                start_gas_left: Gas::empty(),
```

Comment:
So I'm clear, this whole output struct is zero-ed because it is enqueued to public, so from the perspective of the current execution "nothing happens". 

But then generally, in public this gets "unwrapped" to actually call the function, at which point the sequencer provides the correct "start_gas_left" etc (which we will subsequently verify in the kernel).

Is that correct? Can we add a comment if so/otherwise?

Reply:
Yep, that's it! I guess the comment above tried to cover this, but could be expanded a bit.

>        // Note: Not using PublicCirclePublicInputs::deserialize here, because everything below args_hash is 0 and
>        // there is no more data in fields because there is only ENQUEUE_PUBLIC_FUNCTION_CALL_RETURN_SIZE fields!